megamek/src/megamek/common/MiscType.java

```diff
@@ -745,10 +745,20 @@ public double getTonnage(Entity entity, int location) {
             return TestEntity.ceil(weight, roundWeight);
 
         } else if (hasFlag(F_BASIC_FIRECONTROL)) {
+            // Omni support vees have a fixed weight for the chassis, which may be
```

Comment:
This appears to be identical to the logic for advanced fire control, except the weight multiplier near the bottom. Let's put it into a function?

Reply:
I consolidated them in a single block. I also spotted and fixed an error in the logic to exclude certain weapons used by small support vehicles. It's supposed to skip light weapons (e.g. any infantry weapons that aren't flagged as support) but it was excluding all weapons with either the infantry or infantry support flags.